api/interface.go

```diff
@@ -30,9 +31,8 @@ type validator interface {
 }
 
 type generator interface {
-	PSK() ([]byte, error)
-	UUID() (uuid.UUID, error)
 	Timestamp() time.Time
+	UniqueTimestamp() primitive.Timestamp
```

Comment:
Why do we need it?

Reply:
In the existing filters service/document this is how it's saved, there was an issue determining the filterType in the other service because this field it was unable to unmarshal a time that had been saved as a UTC time.time.